src/PaginationResult.php

```diff
@@ -50,10 +50,30 @@ public function __construct($rows, array $meta)
     /**
      * Get iterator of records.
      *
-     * @return \ArrayIterator
+     * @return \Traversable|\ArrayIterator
      */
     public function getIterator()
     {
-        return new \ArrayIterator($this->records);
+        return $this->records instanceof \Traversable ? $this->records : new \ArrayIterator($this->records);
+    }
+
+    /**
+     * Count records.
+     *
+     * @return int
+     */
+    public function count()
+    {
+        // @codeCoverageIgnoreStart
+        if (
+            !$this->records instanceof \Countable
+            && !is_array($this->records)
+            && version_compare(PHP_VERSION, '7.2.0', '>=')
+        ) {
+            // PHP: rfc:counting_non_countables https://wiki.php.net/rfc/counting_non_countables
+            trigger_error('count(): Parameter must be an array or an object that implements Countable', E_USER_WARNING);
```

Comment:
This may be emitted by PHP >= 7.2.0, which means we should return here.

Reply:
?

Reply:
I get it, we should just call `count()`.